crud.lua

```diff
@@ -12,7 +12,7 @@ local select = require('crud.select')
 local truncate = require('crud.truncate')
 local len = require('crud.len')
 local borders = require('crud.borders')
-local sharding_key = require('crud.common.sharding_key')
+local sharding_metadata = require('crud.common.sharding.sharding_metadata')
 local utils = require('crud.common.utils')
 
```

Comment:
I have reverted all commits except first one in a patch series and tests failed:

```
sergeyb@pony:~/sources/MRG/crud$ git describe                                                                                           
0.10.0-1-gde58f4e                                                                                                                       
sergeyb@pony:~/sources/MRG/crud$ .rocks/bin/luatest -v test/                                                                            
"tuple.keydef" module is used                                                                                                           
"tuple.merger" module is used                                                                                                           /home/sergeyb/sources/MRG/crud/crud.lua:15: module 'crud.common.sharding_key' not found:No LuaRocks module found for crud.common.shardin
g_key                                                                                                                                   
        no field package.preload['crud.common.sharding_key']                                                                            
        no file '/home/sergeyb/sources/MRG/crud/crud/common/sharding_key.lua'
        no file '/home/sergeyb/sources/MRG/crud/crud/common/sharding_key/init.lua'
```

Probably the same for other commits.

Reply:
Fixed, thanks, now tests are running, but tests for sharding keys are failing because fetching functions have been moved to sharding metadata module and dependencies between sharding metadata module and sharding key module have be resolved in the next commits.

Reply:
Fixed, now all tests are passed:

```lua
function sharding_key_module.extract_from_pk(space_name, primary_index_parts, primary_key, timeout)
     dev_checks('string', 'table', '?', '?number')

     local sharding_key_as_index_obj, err = sharding_key_module.fetch_on_router(space_name, timeout)
     local sharding_key_as_index_obj, err = require(
                                            'crud.common.sharding.sharding_metadata'
                                            ).fetch_on_router(space_name, timeout)
     ...
end
```

```lua
local _fetch_on_router = locked(function(timeout)
     ...
     for space_name, metadata in pairs(metadata_map) do
         local sharding_key_as_index_obj, err = require(
                                                'crud.common.sharding.sharding_key'
                                                ).internal.as_index_object(space_name,
                                                                           metadata.space_format,
                                                                           metadata.sharding_key_def)
     ...
     end
     ...
end
```

Reply:
For me tests ddl_sharding_func_cache, ddl_sharding_func and ddl_sharding_key broken in commit "Add support of custom sharding func for crud methods".

```
94) ddl_sharding_key.engine:"vinyl".test_update_cache_with_incorrect_key                                                           
...rocks/share/tarantool/cartridge/test-helpers/cluster.lua:177: "localhost:13304": /home/sergeyb/sources/MRG/crud/test/entrypoint/
srv_ddl.lua:175: spaces["customers_G_func"]: redundant key "sharding_func"                                                         
stack traceback:   

93) ddl_sharding_key.engine:"vinyl".test_update_cache                                                                              
...rocks/share/tarantool/cartridge/test-helpers/cluster.lua:177: "localhost:13304": /home/sergeyb/sources/MRG/crud/test/entrypoint/
srv_ddl.lua:175: spaces["customers_G_func"]: redundant key "sharding_func"                                                         
stack traceback:   
```

Reply:
For this commit CI works fine, I suppose it may happen because of `ddl` version. In latest version of `ddl` suppuration of sharding func was added, but in previous versions `sharding_func` field was not allowed in schema and this had led to an [error](https://github.com/tarantool/ddl/blob/master/ddl/check.lua#L1016). Please, correct me if I am wrong.

Reply:
Seems the problem was in ddl version, thanks!